torchtune/models/phi3/_sentencepiece.py

```diff
@@ -0,0 +1,225 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import List, Optional, Tuple
+
+from sentencepiece import SentencePieceProcessor
+from torchtune.data._types import Message
+from torchtune.data._utils import truncate
+
+
+class Phi3MiniSentencePieceTokenizer:
```

Comment:
Wait I just now realized that our sentencepiece tokenizer doesn't inherit from the tokenizer protocol. Is there a reason for this? Cause tiktoken does

Reply:
Bad engineering?